### ISSUE TITLE
Make member-ordering rule more flexible

### ISSUE THREAD

**Issue by JKillian** [palantir/tslint#1208]
**New Options Syntax**

All of the below examples would be supported:

Full customization if needed:

```json
"member-ordering": [true, { "order": [
  "public-static-field",
  "protected-static-field",
  "private-static-field",
  "public-instance-field",
  "protected-instance-field",
  "private-instance-field",
  "constructor",
  "public-static-method",
  "protected-static-method",
  "private-static-method",
  "public-instance-method",
  "protected-instance-method",
  "private-instance-method",
]}]
```

Preset orderings (what most people would use):
```
// would do something similar to above
"member-ordering": [true, { "order": "smart" }]
```

Middle of the road orderings (although I'm debating if this is worth implementing):

```
"member-ordering": [true, { "order": [
  "static-field",        // public before private assumed
  "instance-field",
  "constructor",
  "static-method",
  "instance-method",
]}]
```

and

```
"member-ordering": [true, { "order": [
  "constructor",
  "static",       // field before method assumed and public before private assumed
  "instance",
]}]
```

**Comment by lowkay** [comment]
Yupp this looks like a much neater way to do it. I think your approach to have a good default is a good choice.

I think that middle road presets would add unnecessary complexity to tslint. Start simple and if there's lots of noise think about middle ground.

Looking forward to seeing this implemented :)

**Comment by JKillian** [comment]
👍  on avoiding the middle of the road options. These would add implementation and documentation complexity,  and for the most part the first two options should cover what people need.

Also interested in what presets people would like to see, and what they should be named, if anyone has ideas.

**Comment by JKillian** [comment]
Closed by #1210
